qcodes/instrument_drivers/yokogawa/GS200.py

```diff
@@ -191,11 +191,10 @@ def __init__(self, name: str, address: str, terminator: str = "\n",
                            set_cmd=self._set_source_mode,
                            vals=Enum('VOLT', 'CURR'))
 
-        # When getting the mode internally in the driver, look up the mode as
-        # recorded by the _cached_mode property, instead of calling source
-        # _mode(). This will prevent frequent VISA calls to the instrument.
-        # Calling _set_source_mode will change the chased value.
-        self._cached_mode = "VOLT"
+        # The source_mode attribute of the instrument. The default mode is
+        # VOLT when the instrument is truned on. The mode can be switched
+        # between VOLT and CURR.
+        self.source_mode()
 
         # We want to cache the range value so communication with the instrument
         # only happens when the set the range. Getting the range always returns
```

Comment:
Can we also get rid of cached_range_value just below and `_auto_range ` for the same reasons (use the cache within the parameter)

Reply:
@jenshnielsen 
Regarding auto_range, as you probably know, the instrument does not support this functionality. So, I don't know how to initialize the instrument with the previous state of auto_mode (True or False). However, I can replace self._auto_range=False with initial_cache_value=False in the auto_range parameter.
Is there a way to remember the previous state or in this case, the logic of initializing auto_mode in False state is inescapable?

Reply:
@farbo since the autorange is not build into the instrument we can freely pick the default and set it with `initial_cache_value`
as you suggest. We should not try to keep state for things that are not set on the insturment.

Reply:
Good point. Thanks